cliquet/resource/__init__.py

```diff
@@ -611,6 +611,11 @@ def _add_cache_header(self, response):
         collection_expires = self.request.registry.settings.get(setting_key)
         if collection_expires is not None:
             response.cache_expires(seconds=int(collection_expires))
+        else:
+            # Since Expires has resolution in seconds, do not use cache_expires
+            # in order to omit it.
```

Comment:
Isn't `Expires: 0` equivalent to `Cache-Control: no-cache`? The `no-cache` directive does *not* mean don't cache (despite its name), it just means 'revalidate each time', where 'revalidate' means send a conditional request to the server and check that you get a 403 and not a 401 or 200 before serving the data from cache. So this `if / else` statement creates a situation where adding an `collection_expires` will cause the client to not revalidate. Revalidation is useful for making sure that each client receives updates to the resource contents asap, but also for revoking access. So setting `collection_expires` to a date in the future now has the side-effect that you cannot revoke access to the resource until that date.